### PR TITLE
MGDOBR-774: Cover creating/editation of the Processor

### PR DESCRIPTION
**Pull request by sjamboro**
https://issues.redhat.com/browse/MGDOBR-774

### CODE REVIEW

File: cypress/integration/ProcessorTest.ts [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+            .find("td")
+            .each((cell, cellIndex) => {
+              cy.wrap(cell).should("have.text", filters[rowIndex][cellIndex]);
+            });
+        });
+      });
+
+      cy.get("code").should("have.text", "I want to write here something.");
+
+      cy.get("[class='pf-c-description-list__description']")
```
Comment (jomarko) [inline cypress/integration/ProcessorTest.ts:121]: `pf-c-description-list__description` elements contain `parameter values`. Would you mind if we check also `pf-c-description-list__term` elements, which should contain `parameter keys`?

File: cypress/integration/ProcessorTest.ts [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+      processorName = "Processor two";
+      action = [
+        "Send to Slack",
+        "test",
+        "https://hooks.slack.com/services/XXXXXXXX/XXXXXXXXX/XXXXXXXXXXXXXXXXXXX",
+      ];
+      filters = [["data.name", "processor.StringEquals", "John"]];
+    });
+
+    it.skip("Details Header", () => {
```
Comment (jomarko) [inline cypress/integration/ProcessorTest.ts:253]: could we please document/comment reason of skipping?

File: cypress/integration/ProcessorTest.ts [hunk 3]
```diff
@@ -0,0 +1,342 @@
...
+            .find("td")
+            .each((cell, cellIndex) => {
+              cy.wrap(cell).should("have.text", filters[rowIndex][cellIndex]);
+            });
+        });
+      });
+    });
+  });
+
+  describe("Edit Sink Processors", () => {
```
Comment (jomarko) [inline cypress/integration/ProcessorTest.ts:230]: I think the only supported edit operation, for processors, is a change of processor filters. Please double check it with team. If answer is yes. We should remove some tests, for example edit processor name.

**Review by jomarko — COMMENTED** [review]
Thank you for a PR @sjamboro

File: cypress/integration/ProcessorTest.ts (hunk 3 above)
Comment (vpellegrino) [reply cypress/integration/ProcessorTest.ts:230]: After https://issues.redhat.com/browse/MGDOBR-731 sure, we need to cover with some IT tests about specific filter types.
AFAIK, the processor name change is supported

**Comment by sjamboro** [comment]
@vpellegrino @jomarko PR is ready for review. It covers just base creation and editation of processor (cancel without modification; add filter row). The test suite checks Source and Sink Processor separetely. We can think about deep tests later.

File: cypress/integration/ProcessorTest.ts (hunk 3 above)
Comment (kelvah) [reply cypress/integration/ProcessorTest.ts:183]: Processor name editing was working fine when I reviewed #54. I've tested it against the dev APIs. So something has changed. Let's check this next week.

Comment (kelvah) [reply cypress/integration/ProcessorTest.ts:183]: I checked and I was wrong. Processor names cannot be changed right now.

File: cypress/integration/ProcessorTest.ts [hunk 4]
```diff
@@ -0,0 +1,449 @@
...
+    it("Sink processor", () => {
+      const processorName: string = "Sink processor";
+      const action = [
+        "Send to Slack",
+        "dev-action",
+        "https://test.app.com/item",
+      ];
+      const filters = [
+        ["data.name", "processor.StringEquals", "John"],
+        ["data.surname", "processor.StringEquals", "White"],
```
Comment (jomarko) [inline cypress/integration/ProcessorTest.ts:23]: juts double checking, I think in some other PR there was a discussion `processor.StringEquals` is a consequence of some missing translation? is `processor.StringEquals` string constant still expected at this line?

File: cypress/integration/ProcessorTest.ts [hunk 5]
```diff
@@ -0,0 +1,449 @@
...
+      cy.ouiaId("item-0").within(() => {
+        cy.ouiaId("filter-key", "PF4/TextInput").type(filters[0][0]);
+        cy.ouiaId("filter-type", "PF4/FormSelect").select("String equals");
+      });
+      cy.ouiaId("item-0") //The filter-value was detached from DOM and we need to find the context again.
+        .ouiaId("filter-value", "PF4/TextInput")
+        .type(filters[0][2]);
+      cy.ouiaId("item-1").within(() => {
+        cy.ouiaId("filter-key", "PF4/TextInput").type(filters[1][0]);
+        cy.ouiaId("filter-type", "PF4/FormSelect").select("String equals");
```
Comment (jomarko) [inline cypress/integration/ProcessorTest.ts:45]: related to my previous question , shouldn't we select `filters[1][1]` or sth similar?

File: cypress/integration/ProcessorTest.ts [hunk 6]
```diff
@@ -0,0 +1,449 @@
...
+      cy.ouiaId("processor-name", "PF4/TextInput")
+        .should("be.visible")
+        .type(processorName);
+
+      //Filters
+      cy.ouiaId("add-filter", "PF4/Button").should("be.visible").click();
+
+      cy.ouiaId("item-0").within(() => {
+        cy.ouiaId("filter-key", "PF4/TextInput").type(filters[0][0]);
+        cy.ouiaId("filter-type", "PF4/FormSelect").select("String equals");
```
Comment (jomarko) [inline cypress/integration/ProcessorTest.ts:38]: related to my previous question , shouldn't we select `filters[0][1]` or sth similar?

File: cypress/integration/ProcessorTest.ts [hunk 7]
```diff
@@ -0,0 +1,449 @@
...
+        filters
+      );
+    });
+
+    it("Source processor", () => {
+      const processorName: string = "Source processor";
+      const source = ["Slack", "dev channel", "asd14u-e"];
+      const filters = [
+        ["data.name", "processor.StringEquals", "John"],
+        ["data.surname", "processor.StringEquals", "White"],
```
Comment (jomarko) [inline cypress/integration/ProcessorTest.ts:112]: similar question about missing translations

File: cypress/integration/ProcessorTest.ts [hunk 8]
```diff
@@ -0,0 +1,449 @@
...
+      cy.ouiaId("channel", "PF4/TextInput")
+        .should("be.visible")
+        .type(source[1]);
+      cy.ouiaId("token", "PF4/TextInput").should("be.visible").type(source[2]);
+
+      //Filters
+      cy.ouiaId("add-filter", "PF4/Button").should("be.visible").click();
+      cy.ouiaId("item-0").within(() => {
+        cy.ouiaId("filter-key", "PF4/TextInput").type(filters[0][0]);
+        cy.ouiaId("filter-type", "PF4/FormSelect").select("String equals");
```
Comment (jomarko) [inline cypress/integration/ProcessorTest.ts:140]: similar question as for sink test

File: cypress/integration/ProcessorTest.ts [hunk 9]
```diff
@@ -0,0 +1,449 @@
...
+      });
+      //The filter-value was detached from DOM and we need to find the context again.
+      cy.ouiaId("item-0")
+        .ouiaId("filter-value", "PF4/TextInput")
+        .type(filters[0][2]);
+      cy.ouiaId("item-1").within(() => {
+        cy.ouiaId("filter-key", "PF4/TextInput").type(filters[1][0]);
+        cy.ouiaId("filter-type", "PF4/FormSelect").select("String equals");
```
Comment (jomarko) [inline cypress/integration/ProcessorTest.ts:148]: similar question as for sink test

**Review by jomarko — COMMENTED** [review]
Thank you for updates @sjamboro . Some questions from me about the string constants we use in test.

**Review by jomarko — APPROVED** [review]
Approving, as CI passed after rebase.

**Review by kelvah — APPROVED** [review]
LGTM

**Review by vpellegrino — APPROVED** [review]
LGTM, thank you @sjamboro

**Review by jomarko — APPROVED** [review]
@sjamboro I checked just https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox-ui/pull/68/commits/c91544c528553f9eebf4cd2e38e36727e082a23e

it seems fine to me, please, just be sure we do not commit mp4 videos. Probably .gitignore update needed?

**Pull request merged**